### PR TITLE
1710923: GUI: Do not auto-attach, when golden ticket is used; ENT-1309

### PR DESCRIPTION
**Pull request by jirihnidek**
* BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1702988
* Do not show anything related to auto-attach, when golden ticket is used
* Hide button "Auto-attach", when golden ticket is used
* Added function for detection of golden ticket usage. It could be used
  in other parts of subscription-manager

### CODE REVIEW

File: src/subscription_manager/utils.py [hunk 1]
```diff
@@ -161,6 +161,36 @@ def is_valid_server_info(conn):
         return False
 
 
+def is_owner_using_golden_ticket(uep=None, identity=None, owner=None):
```
Comment (cnsnyder) [inline src/subscription_manager/utils.py]: Can we cache the result of this function? As it is used presently I'm concerned that we'll be increasing the number of calls made to the registered entitlement platform.

Comment (jirihnidek) [reply src/subscription_manager/utils.py]: Sure. It's good point.

File: src/subscription_manager/utils.py [hunk 2]
```diff
@@ -161,6 +161,59 @@ def is_valid_server_info(conn):
...
+
+    # When identity is not known, then system is not registered
+    if identity.uuid is None:
+        return False
+
+    content_access_mode = None
+
+    # We have to load it here, because we don't want to add another class to dependency injection
+
+    # Try to use chached data to minimize numbers of REST API calls
```
Comment (cnsnyder) [inline src/subscription_manager/utils.py]: chached -> cached

Comment (jirihnidek) [reply src/subscription_manager/utils.py]: Done.

File: src/subscription_manager/gui/registergui.py [hunk 3]
```diff
@@ -444,6 +444,10 @@ def choose_initial_screen(self):
             # to subscriptions" is clicked in the gui)
             if self.info.get_property('skip-auto-bind'):
                 self.emit('finished')
+            # When golden ticket mode is used, then skit auto-bind too
+            if is_owner_using_golden_ticket():
+                log.debug('Skipping auto-bind, because contentAccessMode is equal to organization')
```
Comment (cnsnyder) [inline src/subscription_manager/gui/registergui.py]: Shouldn't this be: "Skipping auto-bind, because contentAccessMode is equal to org_environment"

**Comment by cnsnyder** [comment]
@jirihnidek Just checking back in, any update for this PR?

**Comment by jirihnidek** [comment]
@cnsnyder Hi, all typos should be fixed now.

**Pull request merged**